observability/WFLY-17144-add-micrometer-support.adoc

```diff
@@ -0,0 +1,88 @@
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-17144
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1686 - Implement the Observability policy - Metrics (XP)
+* https://issues.redhat.com/browse/EAP7-1805 - Implement the Observability policy - Metrics - Base EAP
+* https://issues.redhat.com/browse/WFLY-17138 - Convert MP Metrics subsystem to Admin-Only Mode
+* https://issues.redhat.com/browse/EAP7-1975 - Remove MP Metrics functionality and convert subsystem to Admin-Only Mode
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+* [X] Engineering
+* [ ] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+* Metrics
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [ ] Managed domain
+* [x] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* The server/domain administrator should be able to configure Micrometer metrics
+* The extension/subsystem will not be added automatically to standard configs and must be installed manually. 
+** This is to prevent the dual gathering of metrics with WildFly Metrics still enabled. T
+** Admins are expected to pick one system or the other and modify local configs as required.
+* Server and JVM metrics tracked and exposed
+* Metrics will be exported via the OTLP protocol to a compatible collector via the default (localhost) endpoint.
+* Availability of the metric registry for injection into user applications
+** CDI injection of the `MeterRegistry`
+** Deployed applications will be able to register metrics for export alongside the various server metrics via the Micrometer APIs
+
+=== Non-Requirements
+
+* Support for multiple metric backends (e.g., Datadog, New Relic, Prometheus)
+* Automatic timing of REST requests
+* Removal of MicroProfile Metrics (This will be handle via another WildFly JIRA)
+* Removal of application metrics when a user application is undeployed
+* Mechanism for arbitrary subsystems to register meters (e.g., Micrometer's `MeterBinder`)
+
+== Backwards Compatibility
+
+There are no backwards compatibility issues with this new extension.
+
+== Security Considerations
+
+When pushing metrics to the collector, there is no user involved from which to determine credentials/permissions, so the normal RBAC rules cannot be applied directly. The typical Micrometer deployment involves pushing all metrics, irrespective of permissions, to the collector, which then exports the metrics to one or more observability platforms, which are then secured in a manner specific to those systems. The operation to read the metrics values from the WildFly model, though, will be run using the `Monitor` role to restrict what metrics are pushed to those that should be monitored.
```

Comment:
regarding `operation`, is that about the JBoss CLI?

Reply:
Thanks for rewording this.

Reply:
Of course! :)